### ISSUE TITLE
versions of dependencies

### ISSUE THREAD

**Issue by rmandelb** [GalSim-developers/GalSim#841]
@rmjarvis - can we give any further guidance in INSTALL.md regarding the necessary versions of GalSim's various dependencies?  This came up in the Euclid context, but I think it is kind of useful more generally.

We mention required versions for python, scons, FFTW, and TMV in INSTALL.md.  Can we add anything useful about required versions for NumPy, boost, pyfits or astropy, future, pyyaml, or the optional dependencies?

**Comment by rmandelb** [comment]
@brgillis - I opened an issue for this to help fill in that info, JFYI.

**Comment by rmjarvis** [comment]
I think we're compatible with rather old versions of numpy, boost, and pyfits/astropy, since we've had lots of systems over the years that had various old versions of these and added workaround for various bugs we discovered in them.  However, we probably don't want to guarantee compatibility with them, since I'm not sure which old versions are still semi-regularly tested.

For a "minimum requirements" listing, it would probably be safer to take whatever the current versions used by Travis and put a statement like "GalSim has been tested to work with the following versions, but it is likely that most versions released within the last several years will also work."

**Comment by rmandelb** [comment]
Thanks.  That's a good way to deal with it.  While this is just a documentation change, I still want to do it on a branch and make a PR just so you can easily view the changes and make sure I'm not missing something.  Will do it in a sec.

**Comment by rmandelb** [comment]
Actually... is it just me or do we not have a good way to get Travis to tell us which NumPy, future, or pyyaml version it's using?  I was looking at the logs, like https://travis-ci.org/GalSim-developers/GalSim/jobs/183439328 , but they don't seem to say the versions for those.

The logs do give versions for boost and astropy.  I can modify the build process to spit out the versions used and pull that info from the next build, but is there another way to get this information?

**Comment by rmjarvis** [comment]
Numpy seems to be pre-installed, so we don't get the report from pip.  I'm pretty sure it's 1.6.1 though.  I could add a print statement to SCons that will tell us.

The rest though are reported here:
```
Successfully installed argparse-1.4.0 astropy-1.1.1 codecov-2.0.5 coverage-4.2 coveralls-1.1 cycler-0.10.0 docopt-0.6.2 future-0.16.0 matplotlib-1.5.2 pyparsing-2.1.10 python-dateutil-2.6.0 pytz-2016.10 pyyaml-3.12 requests-2.12.3 scipy-0.17.0 starlink-pyast-3.9.0
```

**Comment by rmandelb** [comment]
Thanks; I missed that (obviously).  Adding a print statement to scons might be useful for other reasons.  Want me to do it on this branch?

Regarding the rest, I had one more question/thought.  The INSTALL.md file is a bit overwhelming in the optional dependencies it reports.  Could we make a statement about "here's the minimum you need to get nearly all the GalSim functionality"?  For example, if someone uses astropy rather than pyfits, they automatically get WCS functionality from astropy.wcs and don't need to separately install starlink etc. (but they do need scipy...) It seems to me that the minimal list is as follows:

numpy, astropy, scipy (only if you want to use astropy.wcs, otherwise you don't need it), future, scons, fftw, tmv, boost

If someone wants to use yaml for config, then they also need pyyaml.
If someone wants faster text file parsing for bandpasses, etc., they also need pandas (but it'll work more slowly without pandas).

Is this a good summary?  All that info is distributed throughout the INSTALL.md but I thought having a brief summary of all this would be useful.

**Comment by rmjarvis** [comment]
I added statements from Scons about the versions that it can determine.  (I couldn't find a way to get the fftw version programatically.)

Fine with any streamlining you want to do in INSTALL.  I agree it is rather unwieldy at the moment.  The only change to your summary I'd make is moving scipy to be in the optional list with yaml, pandas.  

Even astropy is technically optional, although probably everyone uses that for pyfits now, so maybe we should switch to just making astropy officially required.

**Comment by rmandelb** [comment]
Thanks for adding those statements.  Do you know how I can force travis to recognize that it needs to rebuild on this branch so I can grab those version numbers?  Do I have to make a PR?

> The only change to your summary I'd make is moving scipy to be in the optional list with 
> yaml, pandas.

Yes, fair enough.

> Even astropy is technically optional

That's true, but my thinking when constructing minimal recommendations is that we should try to get maximal functionality.  If we just do pyfits, then there's nothing for WCS and the chromatic functionality won't work since it needs astropy units.

I don't object to making astropy required, though I'm also OK with keeping things as-is.

**Comment by rmjarvis** [comment]
> If we just do pyfits, then there's nothing for WCS 

Au contraire.  GSFits works for almost all normal FITS files.  You only need starlink (preferred) or astropy for some of the more esoteric FITS WCS options.

**Comment by rmandelb** [comment]
My bad.  Correction: if we do pyfits, then we can't use chromatic functionality (I thought it needs astropy units).  So we'd have to have pyfits plus astropy.units, at least, even if not using astropy.io.fits.

**Comment by rmjarvis** [comment]
> So we'd have to have pyfits plus astropy.units, at least, even if not using astropy.io.fits.

Right.  Probably easiest to just recommend people install astropy >= 1.2 and not be so nuanced about it.  (Really >= 1.1, but 1.1.2 is broken, so we don't want them to get that.)

**Comment by rmandelb** [comment]
Hmmm, the travis system seems to be doing 1.1.1, so I put that.

**Comment by rmandelb** [comment]
And this comment may have gotten lost, so let me just check again: how can I force travis to rebuild in this branch so i can pull out the numpy version it's using?  That's the last thing I am waiting on before making a PR.

**Comment by rmjarvis** [comment]
Right.  You can't.  It only builds for master and things being proposed for merge into master.

**Comment by rmandelb** [comment]
OK.  Will make the PR and ask you to hold off until it's done the build and I've done the final edit to INSTALL.md.  Thanks!

**Comment by rmandelb** [comment]
Closing #841.
